### PR TITLE
[DON'T MERGE] jUnit test for mwm sizes;

### PR DESCRIPTION
**Pull request by therearesomewhocallmetim**
This is a junit test that compares mwm sizes in the current countries.txt and a file with the previous mwm sizes that must be called old_countries.txt.

### CODE REVIEW

File: tools/java_tests/src/test/java/me/maps/utils/Utils.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+
+public class Utils {
+	public static final String VERSION = "__version__";
+
+
+	public static Map<String, Long> readFilesizesFromCoutriesTxt(String path) {
```
Comment (kshalnev) [inline tools/java_tests/src/test/java/me/maps/utils/Utils.java]: может не объединять размеры mwm и не mwm.routing, а процессить каждый из них отдельно?

File: tools/java_tests/build.gradle [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    compile 'org.json:json:20150729'
+
+    testCompile 'junit:junit:4.12'
+    testCompile 'com.jayway.restassured:rest-assured:2.7.0'
+}
+
+
+task showMeCache << {
+  configurations.compile.each { println it }
+}
```
Comment (syershov) [inline tools/java_tests/build.gradle]: \n

File: tools/java_tests/src/test/java/me/maps/utils/Utils.java (hunk 1 above)
Comment (ygorshenin) [inline tools/java_tests/src/test/java/me/maps/utils/Utils.java]: readFileSizesFromCountriesFile

File: tools/java_tests/src/test/java/me/maps/utils/Utils.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+
+public class Utils {
+	public static final String VERSION = "__version__";
```
Comment (ygorshenin) [inline tools/java_tests/src/test/java/me/maps/utils/Utils.java]: Two spaces indent, please.

File: tools/java_tests/src/test/java/me/maps/utils/Utils.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
+package me.maps.utils;
+
+
```
Comment (ygorshenin) [inline tools/java_tests/src/test/java/me/maps/utils/Utils.java]: Remove redundant blank lines, please.

File: tools/java_tests/src/test/java/me/maps/utils/Utils.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import org.json.JSONObject;
+
+
+public class Utils {
+	public static final String VERSION = "__version__";
+
+
+	public static Map<String, Long> readFilesizesFromCoutriesTxt(String path) {
+		try {
+			List<String> newCountries = Files.readAllLines(Paths.get(path));
```
Comment (ygorshenin) [inline tools/java_tests/src/test/java/me/maps/utils/Utils.java]: s/newCountries/contents/

File: tools/java_tests/src/test/java/me/maps/utils/Utils.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+			String s = newCountries.stream().collect(Collectors.joining());
```
Comment (ygorshenin) [inline tools/java_tests/src/test/java/me/maps/utils/Utils.java]: s -> contents

File: tools/java_tests/src/test/java/me/maps/tests/MwmSizeTest.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+public class MwmSizeTest {
+
+	public static Map<String, Long> oldMwmSizes;
+	public static Map<String, Long> newMwmSizes;
+
+
+	@BeforeClass
+	public static void readMwmSizes() throws Exception {
+		newMwmSizes = readFilesizesFromCoutriesTxt("../../data/countries.txt");
```
Comment (syershov) [inline tools/java_tests/src/test/java/me/maps/tests/MwmSizeTest.java]: Can we define this in settings file?

**Comment by ygorshenin** [comment]
One question to the whole PR: why Java, not Python? I guess on Python this code will be 10x times shorter and clearer.

**Comment by therearesomewhocallmetim** [comment]
@ygorshenin Re "One question to the whole PR": The idea is to make not just one test, but a scalable and manageable test suite. It is possible to make this 10x shorter in Java as well, but it would be difficult to add more tests to it. Currently, there are more tests in the pipeline relying on the infrastructure established by this piece of code. 
Also, my Java is better than my Python, and the Python tests I have seen were not that good.

**Comment by gardster** [comment]
Maybe it will be better to use python and pyUnit?

**Comment by therearesomewhocallmetim** [comment]
I will look into PyUnit, I have some reservations, though.

File: tools/java_tests/src/test/java/me/maps/matchers/MwmMapMatcher.java [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+	}
+
+
+	@Override
+	protected boolean matchesSafely(Map<String, Long> map) {
+		newValues = map;
+
+		boolean passed = true;
+
+		Set<String> newKeys = new TreeSet<String>(newValues.keySet());
```
Comment (yunikkk) [inline tools/java_tests/src/test/java/me/maps/matchers/MwmMapMatcher.java]: Обсудили в ЛС - логика слишком сложная, достаточно было бы сравнить размеры new-old values и проитерироваться по всем элементам. Впрочем - если логика работает и упрощать не хочется, поставлю LGTM/

**Comment by Zverik** [comment]
Кстати, в мастере уже пару дней лежит [аналогичный скрипт на питоне](https://github.com/mapsme/omim/blob/master/tools/unix/diff_size.py), только сравнивает не по countries, а сами файлы. И вызывается из `test_planet.sh`.

File: tools/java_tests/src/test/resources/log4j.xml [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+			<param name="ConversionPattern" value="%d{yyyy-MM-dd HH:mm:ss} %-5p %c{1}:%L - %m%n" />
+		</layout>
+	</appender>
+
+	<root>
+		<level value="TRACE" />
+		<appender-ref ref="console" />
+	</root>
+
+</log4j:configuration>
```
Comment (Zverik) [inline tools/java_tests/src/test/resources/log4j.xml]: Перевод строки в конце потерялся.

File: tools/java_tests/build.gradle [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+task buildApplication(type: GradleBuild) {
+    buildFile = '../../android/build.gradle'
+    dir = "../../android"
+    tasks = ['assembleGoogleRelease']
+}
+
+
+
+ task deviceTests (type: Test, dependsOn: [testClasses]) {//, obbGenerate, appClean, buildApplication]) {
+// task deviceTests (type: Test, dependsOn: [testClasses, obbGenerate, appClean, buildApplication]) {
```
Comment (mapsmetest) [inline tools/java_tests/build.gradle]: Uncomment this line and comment/remove the previous one.

**Comment by deathbaba** [comment]
@therearesomewhocallmetim этот реквест еще нужен?

**Pull request closed without merging**